test_submission_schema/schemas/v1.json

```diff
@@ -0,0 +1,185 @@
+                            "description": "The name of the package installed on the system.",
+                            "examples": ["yarf", "checkbox"]
+                        },
+                        "version": {
+                            "type": "string",
+                            "description": "The version of the package installed on the system.",
+                            "examples": ["1.0.0"]
+                        },
+                        "revision": {
+                            "type": "string",
+                            "description": "The revision number of the installed Snap.",
+                            "examples": ["100", "150"]
+                        }
+                    },
+                    "if": {"properties": {"type": {"const": "snap"}}},
+                    "then": {"required": ["type", "name", "version", "revision"]},
+                    "else": {
+                        "if": {"properties": {"type": {"const": "debian"}}},
+                        "then": {"required": ["type", "name", "version"]},
+                        "else": {"required": ["type", "version"]}
+                    }
+                }
+            }
+        },
+        "session_data": {
+            "type": "object",
+            "required": [
+                "title",
+                "test_plan_id"
+            ],
+            "properties": {
+                "title": {
+                    "type": "string",
+                    "description": "The title of the test plan."
+                },
+                "test_plan_id": {
+                    "type": "string",
+                    "pattern": "^[a-zA-Z0-9_]+\\.[a-zA-Z0-9_]+\\.[a-zA-Z0-9_]+::[a-zA-Z0-9_]+$",
+                    "description": "The unique identifier of the test plan.",
+                    "examples": ["com.canonical.yarf::dell_bluetooth", "com.canonical.checkbox::hp_audio"]
+                },
+                "description": {
+                    "type": "string",
+                    "description": "A brief description of the test plan."
+                },
+                "execution_id": {
+                    "type": "string",
+                    "description": "Execution ID for running the test plan."
+                }
+            }
+        },
+        "results": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "required": [
+                    "id",
+                    "test_description",
+                    "certification_status",
+                    "category_id",
+                    "outcome",
+                    "comments",
+                    "io_log",
+                    "duration",
+                    "type"
+                ],
+                "properties": {
+                    "id": {
+                        "type": "string",
+                        "pattern": "^[a-zA-Z0-9_]+\\.[a-zA-Z0-9_]+\\.[a-zA-Z0-9_]+::[a-zA-Z0-9_]+/[a-zA-Z0-9_]+$",
```

Comment:
Is it a requirement to include exactly two segments? Some existing test cases already has `/` in their ID. Also, hyphen `-` and period `.` are already in use, as in `info/hdparm_name.txt`

Reply:
From discussion in MM, let's make it free form for now.